### PR TITLE
feat(copy/backport): add bot_account option

### PR DESCRIPTION
**Pull request by sileht**
Fixes MRGFY-406

### CODE REVIEW

**Review by jd — DISMISSED** [review]
doc?

File: docs/source/actions/backport.rst [hunk 1]
```diff
@@ -62,6 +62,13 @@ Options
      - ``true``
      - Whether to create the pull requests even if they are conflicts when
        cherry-picking the commits.
+   * - ``bot_account``
+     - :ref:`data type template`
+     -
+     - |premium plan tag|
+       Mergify can impersonate a GitHub user to backport a pull request.
+       If no ``bot_account`` is set, Mergify will backport the pull request
+       itself.
```
Comment (jd) [inline docs/source/actions/backport.rst:71]:
```suggestion
       If no ``bot_account`` is set, Mergify backports the pull request
       itself.
```

File: docs/source/actions/copy.rst [hunk 2]
```diff
@@ -42,6 +42,13 @@ Options
      - ``true``
      - Whether to create the pull requests even if they are conflicts when
        cherry-picking the commits.
+   * - ``bot_account``
+     - :ref:`data type template`
+     -
+     - |premium plan tag|
+       Mergify can impersonate a GitHub user to copy a pull request.
+       If no ``bot_account`` is set, Mergify will copy the pull request
```
Comment (jd) [inline docs/source/actions/copy.rst:50]:
```suggestion
       If no ``bot_account`` is set, Mergify copies the pull request
```

File: mergify_engine/duplicate_pull.py [hunk 3]
```diff
@@ -254,15 +255,28 @@ async def duplicate(
...
 
+    bot_account_token: typing.Optional[str] = None
+    if bot_account is not None:
+        user_tokens = await ctxt.repository.installation.get_user_tokens()
+        bot_account_token = user_tokens.get_token_for(bot_account)
+        if not bot_account_token:
+            raise DuplicateFailed(
+                f"{kind} fail: user `{bot_account}` is unknown. "
+                f"Please make sure `{bot_account}` has logged in Mergify dashboard."
+            )
```
Comment (jd) [inline mergify_engine/duplicate_pull.py:266]: Do we miss the plan feature check?

**Review by jd — CHANGES_REQUESTED** [review]

File: mergify_engine/duplicate_pull.py (hunk 3 above)
Comment (sileht) [reply mergify_engine/duplicate_pull.py:266]: This is done in the action.

**Review by jd — APPROVED** [review]

**Pull request merged**